frontend/src/app/main/data/workspace/groups.cljs

```diff
@@ -147,29 +148,25 @@
              (filter #(#{(:id group)} (second %)))
              (ffirst))
 
-        rchanges [{:type :mov-objects
-                   :page-id page-id
-                   :parent-id parent-id
-                   :shapes shapes
-                   :index index-in-parent}
-                  {:type :del-obj
-                   :page-id page-id
-                   :id (:id group)}]
-        uchanges [{:type :add-obj
-                   :page-id page-id
-                   :id (:id group)
-                   :frame-id (:frame-id group)
-                   :obj (assoc group :shapes [])}
-                  {:type :mov-objects
-                   :page-id page-id
-                   :parent-id (:id group)
-                   :shapes shapes}
-                  {:type :mov-objects
-                   :page-id page-id
-                   :parent-id parent-id
-                   :shapes [(:id group)]
-                   :index index-in-parent}]]
-    [rchanges uchanges]))
+        ids-to-detach (when (:component-id group)
+                        (cp/get-children (:id group) objects))
+
+        detach-fn (fn [attrs]
+                    (dissoc attrs
+                            :component-id
+                            :component-file
+                            :component-root?
+                            :remote-synced?
+                            :shape-ref
+                            :touched))]
+
+    (cond-> (-> (cb/empty-changes it page-id)
+                (cb/with-objects objects)
+                (cb/change-parent parent-id children index-in-parent)
+                (cb/remove-objects [(:id group)]))
+
+      (some? ids-to-detach)
+      (cb/update-shapes ids-to-detach detach-fn))))
```

Comment:
This can be written in this way that I think is easier to read:

```clojure
(-> (cb/empty-changes it page-id)
    (cb/with-objects objects)
    (cb/change-parent parent-id children index-in-parent)
    (cb/remove-objects [(:id group)])
    (cond-> (some? ids-to-detach)
      (cb/update-shapes ids-to-detach detach-fn)))
```